flax/serialization.py

```diff
@@ -125,27 +126,18 @@ def _restore_dict(xs, states):
 
 
 def _namedtuple_state_dict(nt):
-  return {'name': nt.__class__.__name__,
```

Comment:
What's the motivation to also change the `state_dict` structure for fixing this?

Reply:
Having the structure like this is inconsistent with the rest of state_dict where we only store data (PyTree leaves) not structure. Also if you use the original NamedTuple you already have the name and fields so you don't have to store them in the state dict as well.

Reply:
Great, thanks!